src/App.tsx

```diff
@@ -8,7 +8,57 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [searchType, setSearchType] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const formattedQuery = query.trim().toLowerCase();
+
+  const filter = (type: string) => {
+    const findQuery = (queryToSearch: string, todos: Todo[]) => {
+      return todos.filter((todo) => (
+        todo.title.toLowerCase().includes(queryToSearch)
+      ));
+    };
+
+    switch (type) {
+      case 'all':
+        return findQuery(formattedQuery, initialTodos);
+
+      case 'active': {
+        const activeTodos = initialTodos.filter((todo) => !todo.completed);
+
+        return findQuery(formattedQuery, activeTodos);
+      }
+
+      case 'completed': {
+        const completedTodos = initialTodos.filter((todo) => todo.completed);
+
+        return findQuery(formattedQuery, completedTodos);
+      }
+
+      default: return initialTodos;
+    }
+  };
+
+  const fetchGoods = () => (
+    getTodos()
+      .then((todo) => setInitialTodos(todo))
+      .catch(() => 'An error occured when loading todos'));
```

Comment:
what does this catch do?
I think better will be set an error to state and show for user